runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/ConfigureTask.java

```diff
@@ -128,31 +120,11 @@ public Void call() throws Exception
         {
             configText = CONFIG_TEXT_DEFAULT;
         }
-        else if ("http".equals(configURL.getProtocol()) || "https".equals(configURL.getProtocol()))
-        {
-            HttpClient client = HttpClient.newBuilder()
-                .version(HTTP_2)
-                .followRedirects(NORMAL)
-                .build();
-
-            HttpRequest request = HttpRequest.newBuilder()
-                .GET()
-                .uri(configURL.toURI())
-                .build();
-
-            HttpResponse<String> response = client.send(
-                request,
-                BodyHandlers.ofString());
-
-            configText = response.body();
-
-        }
         else
         {
-            URLConnection connection = configURL.openConnection();
-            try (InputStream input = connection.getInputStream())
+            try
             {
-                configText = new String(input.readAllBytes(), UTF_8);
```

Comment:
nice thinking 👍 ok, will update

Reply:
@jfallows am trying to think about how this would really work after a `NamespaceConfig` is already resolved from the JSON string.

My understanding is that the `keys` URL should be resolve during the time when the `configText` is been adapted to a `NamespaceConfig`. In other words, the `keys` should be resolved during the following call:

```java
NamespaceConfig namespace = jsonb.fromJson(configText, NamespaceConfig.class);
```

and not afterwards. Because if its done after the JSON string is already adapted to a `NamespaceConfig`, then the keys would have already been resolved. So I do not see how the `readURL` callback would be helpful in that case. Can you please clarify?

Reply:
> Then in `JwtGuardHandler` constructor, we can use `readURL` to resolve the keys.

@Alfusainey ultimately the `JwtGuardHandler` needs the resolved keys, but it is not a requirement to resolve them during JSON parsing.

Reply:
ok, i see 👍